docs/use_schema.md

```diff
@@ -69,6 +69,9 @@ Arrays (i.e. list of values)
 * uuid_array
 * text_array
 
+Layout (static content, not an input): 
+* header: will show the 'name' of your column as header, and optionally description below. Can be used to partition your forms/reports.
```

Comment:
stange sentence, what are you trying to say there , maybe add a screenshot to make things clear

Reply:
'header' should be 'heading' in docs.

Reply:
> Code seams fine, few small suggestions.
> 
> However i'm really not sure about the the whole idea of having a 'special' column type that is not actually a data base column ( brings back memories of the compound horror). Now each time you use a column you need to keep in mind that is might not actually be a column ( same with compound) these types of designs generally break the whole OO-design thing. I would rather have discussed upfront and think about adding elements just to the form ( without adding them to the table as a 'column') . What if a want to add a line or a box or a folded section to a form are these also going to be columns ? does every form element fit into a linear list ?

Do you have suggestion for an alternative solution that is better than this one? An improvement I considered is tot rename 'Column' to 'FormElement' and then use some OO instead of isHeading() to easily differentiate but I thought that refactor be too much.

I personally hated compounds because it destroyed our APIs. So therefore explicitly didn't go that route. Moreover, in graphql headings are filtered out indeed. 

Your point on 'other design elements': I would want to wait until it happens before we refactor. However, in current design those would be implemented as extended types, e.g. LINE(HEADING), START_FOLD(HEADING), END_FOLD(HEADING), analogous to ONTOLOGY(REF) and ONTOLOGY_ARRAY(REF_ARRAY), see -emx2/ColumnTypes.java. And then indeed 'line' and 'start/end section' might not be heading but something else . Therefore it originally was named 'CONSTANT' . 

I considered all these things but thougt it to be premature abstraction and would postpone these changes until we need them.

Reply:
P.S. we also will use this to influence default RECORD view layout (that is the full page card), adding headings in that view. So the fact that we should consider to use it everywhere is for me leading principle to add this at the heart of the domain model.